client/src/app.js

```diff
@@ -203,6 +212,12 @@ export default function main({ DOM, HTTP, route, storage, scanner: scan$, search
         // use an empty object if the map fails loading for any reason
         .merge(extractErrors(HTTP.select('asset-map')).mapTo({}))
 
+  // Assets List State      
+  , assetList$ = !process.env.ASSETLIST_URL ? O.of([]) : reply('assetlist')
+  
+  // Get total Assets from API. If undefined, use 492 as default
+  , totalAssets$ = dropErrors(HTTP.select('assetlist')).map(resp => resp.headers['x-total-results'] || 492)
```

Comment:
I would combine `assetList$` and `totalAssets$` into a single stream of `reply('assetlist').map(r => ({ assets: r.body, total: resp.headers['x-total-results'] }))`

Reply:
This will not work because reply return response.body

reply = (cat, raw) => dropErrors(HTTP.select(cat)).map(r => raw ? r : (r.body || r.text))

Reply:
Changes have been made...

Reply:
See https://github.com/Blockstream/esplora/pull/278#discussion_r530674311

Reply:
😇